dali/pipeline/pipeline.h

```diff
@@ -310,6 +311,38 @@ class DLL_PUBLIC Pipeline {
     }
   }
 
+  /**
+   * @brief Set if the DALI pipeline should create checkpoints between the epochs
+   *
+   * @param enable_memory_stats If checkpoints should be created
+   */
+  DLL_PUBLIC void EnableCheckpointing(bool checkpointing = true) {
```

Comment:
If you'd like a default, let's make it false as it is experimental feature. And the default in the executor is false

Reply:
Here I mirrored the memory stats methods. Checkpointing is disabled by default anyways, because [it is called with an argument](https://github.com/NVIDIA/DALI/pull/5061/files#diff-99a3c2b3a4b0a9f56a3a1f4d76ad18c524941e8db70da46b6c5ba531891fc685R716). I think it just makes more sense in terms of C++ API to make `EnableCheckpointing()` actually enable checkpointing.